lib/graphql/static_validation/rules/fragments_are_used.rb

```diff
@@ -38,7 +38,7 @@ def add_errors(context, used_fragments, defined_fragments)
       end
 
       def find_difference(fragments, allowed_fragment_names)
-        fragments.select {|f| !allowed_fragment_names.include?(f.name) }
+        fragments.select {|f| f.name && !allowed_fragment_names.include?(f.name) }
```

Comment:
:clap: :clap: on working with StaticValidation! There should be some kind of badge for dealing with this code.

This change (required, totally the right thing to do) is such a code smell. In this design, they're order-dependent in ways that are hard to understand and hard to test. Hoping to fix that one day 😖